### ISSUE TITLE
blipbuf integration isn't really doing anything

### ISSUE THREAD

**Issue by Powerlated** [mgba-emu/mgba#2448]
blipbuf is usually integrated into emulators to process amplitude changes that occur **between** output samples without aliasing. 

mGBA's GB and GBA cores seem to be forwarding amplitude changes to blipbuf at timestamps aligned to output samples, making the library practically do nothing in terms of reducing aliasing. 

Either blipbuf could be removed entirely for a small performance boost without any decrease in sound quality, or its usage can be fixed by giving it amplitude changes at the exact cycle timestamps when PSGs update, audio timers overflow, etc.

**Comment by endrift** [comment]
Yeah, I'm aware of this. It can be used better for the GB core but the GBA core doesn't actually do cycle-perfect timing--it samples every 512 (or 256) cycles.
